sync/block_listener.go

```diff
@@ -43,7 +44,7 @@ func NewBlockListener(net service.Service, sync *Syncer, concurrency int, logger
 		Log:                  logger,
 		semaphore:            make(chan struct{}, concurrency),
 		exit:                 make(chan struct{}),
-		receivedGossipBlocks: net.RegisterGossipProtocol(config.NewBlockProtocol),
+		receivedGossipBlocks: net.RegisterGossipProtocol(config.NewBlockProtocol, priorityq.High),
```

Comment:
I'm a little bit afraid about security as an attacker could publish many blocks and prevent us from sending our hare message on time. I wonder if blocks sent during hare would be considered invalid so they won't be propagated and get into this queue. though I guess if you we're eligible for this epoch you could send them whenever you want.. @barakshani, your opinion ?

Reply:
He could have done this even before the priority queue. Actually the priority Q limits that ability to some degree.
In any case, security is not a first prio right now. It is good to open issues describing these concerns, however.

Reply:
I agree that Hare messages of current layer `i` should be prioritized over blocks of layer != i

Reply:
So currently blocks are always prioritized above hare, is that sufficient for now @barakshani ?
@gavraz not focusing on security does not mean we should intendantly introduce security problems as solutions to problems we have. as this change isn't a configuration thing we could just change later it's a feature we want to rely on. I want to make sure we're doing it right instead of creating more debt because of rushing.

Reply:
yes

Reply:
I don't say we should ignore security. I say we are not adding anything major. Before the PR a neighbor could spam us causing messages to be delayed. After this PR a neighbor can still spam us and delay messages. If there is something new here let me know and we decide what to do with Barak.